<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QUdpApp</class>
 <widget class="QWidget" name="QUdpApp">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>637</width>
    <height>561</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>QUdpApp</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>40</y>
     <width>231</width>
     <height>151</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="title">
    <string>发送端</string>
   </property>
   <widget class="QLineEdit" name="SourcePort">
    <property name="geometry">
     <rect>
      <x>87</x>
      <y>21</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="placeholderText">
     <string>端口号可随意设置1000以上</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>66</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Source Port</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="TargetIPSign">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>50</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="placeholderText">
     <string>Target IP Address</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="SingleIP">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>61</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>单播</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="MulIP">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>组播</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="BroadcastIP">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>61</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>广播</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="TargetIPMul">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>80</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="placeholderText">
     <string>Target IP Address</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="TargetBroadcastIP">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>110</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="placeholderText">
     <string>Target IP Address</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="clear_all">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>61</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>清除</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>40</y>
     <width>241</width>
     <height>151</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="title">
    <string>接收端</string>
   </property>
   <widget class="QLineEdit" name="TargetPort">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>30</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Target Port</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="recvCb">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>49</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>接收</string>
    </property>
   </widget>
   <widget class="QComboBox" name="MulBr_Group">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="Join">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>70</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>加入</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Exit">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>110</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>退出</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="JoinMulBr_IP">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="placeholderText">
     <string>加入组播Source IP</string>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>190</y>
     <width>48</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>接收区：</string>
   </property>
  </widget>
  <widget class="QListWidget" name="recvList">
   <property name="geometry">
    <rect>
     <x>72</x>
     <y>213</y>
     <width>481</width>
     <height>151</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>370</y>
     <width>48</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>发送区：</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="sendMesEdit">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>390</y>
     <width>481</width>
     <height>100</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>100</height>
    </size>
   </property>
   <property name="undoRedoEnabled">
    <bool>true</bool>
   </property>
   <property name="acceptRichText">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="clearsendEditBt">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>490</y>
     <width>80</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>清除发送区</string>
   </property>
  </widget>
  <widget class="QPushButton" name="clearRecvListBt">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>490</y>
     <width>80</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>清除接收区</string>
   </property>
  </widget>
  <widget class="QPushButton" name="sendBt">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>490</y>
     <width>80</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>发送</string>
   </property>
  </widget>
  <widget class="QPushButton" name="test_file">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>490</y>
     <width>80</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Test File</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="version">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>10</y>
     <width>141</width>
     <height>22</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
   <property name="placeholderText">
    <string>显示版本号</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>clearRecvListBt</sender>
   <signal>clicked()</signal>
   <receiver>recvList</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearsendEditBt</sender>
   <signal>clicked()</signal>
   <receiver>sendMesEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>455</y>
    </hint>
    <hint type="destinationlabel">
     <x>145</x>
     <y>410</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
